packages/kiota_abstractions/lib/src/request_configuration.dart

```diff
@@ -2,11 +2,13 @@ part of '../kiota_abstractions.dart';
 
 /// Request configuration type for [BaseRequestBuilder]s.
 class RequestConfiguration<T extends AbstractQueryParameters> {
+
+  RequestConfiguration({required this.queryParameters});
   /// The HTTP headers of the request.
   HttpHeaders headers = HttpHeaders();
 
   /// The request options.
   List<RequestOption> options = [];
 
-  late T queryParameters;
+  T queryParameters;
```

Comment:
Ideally, you should try to make all properties `final`, because then the constructor can be made `const`, which will help the Dart runtime optimize usages of this object.

Reply:
Updated to const constructor by Joanne